dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/icon/IconControllerTest.java

```diff
@@ -60,88 +65,182 @@ void shouldCreateCustomIconWhenFileResourceExist() throws IOException {
+    createCustomIcon(fileResourceId1, keyword, iconKey1);
 
-    JsonObject response = DELETE(String.format("/icons/%s", iconKey)).content();
+    String fileResourceId2 = createFileResource();
+    createCustomIcon(fileResourceId2, keyword, iconKey2);
+
+    String fileResourceId3 = createFileResource();
+    createCustomIcon(fileResourceId3, keyword, iconKey3);
+
+    JsonObject iconResponse = GET("/icons?type=CUSTOM&page=1&pageSize=2").content(HttpStatus.OK);
+    JsonPage pager = iconResponse.get("pager", JsonPage.class);
+
+    JsonList<JsonIcon> icons = iconResponse.getList("icons", JsonIcon.class);
+
+    assertHasMember(iconResponse, "pager");
+    assertHasMember(pager, "total");
+    assertHasMember(pager, "pageSize");
+    assertHasMember(pager, "pageCount");
+
+    assertEquals(3, pager.getTotal());
+    assertEquals(2, pager.getPageSize());
+    assertEquals(2, pager.getPageCount());
+
+    assertEquals(
+        2,
+        icons.size(),
+        () -> String.format("mismatch in number of expected Icon(s), fetched %s", icons));
+  }
+
+  @Test
+  void shouldGetIconsWithDefaultPager() throws IOException {
 
-    assertEquals(String.format("Icon %s deleted", iconKey), response.getString("message").string());
+    String fileResourceId1 = createFileResource();
+    createCustomIcon(fileResourceId1, keyword, iconKey1);
+
+    String fileResourceId2 = createFileResource();
+    createCustomIcon(fileResourceId2, keyword, iconKey2);
+
+    String fileResourceId3 = createFileResource();
+    createCustomIcon(fileResourceId3, keyword, iconKey3);
+
+    JsonObject iconResponse = GET("/icons?type=CUSTOM").content(HttpStatus.OK);
+    JsonPage pager = iconResponse.get("pager", JsonPage.class);
+
+    JsonList<JsonIcon> icons = iconResponse.getList("icons", JsonIcon.class);
+
+    assertHasMember(iconResponse, "pager");
+    assertHasMember(pager, "total");
+    assertHasMember(pager, "pageSize");
+    assertHasMember(pager, "pageCount");
+
+    assertEquals(3, pager.getTotal());
+    assertEquals(50, pager.getPageSize());
+    assertEquals(1, pager.getPageCount());
+
+    assertEquals(
+        3,
+        icons.size(),
+        () -> String.format("mismatch in number of expected Icon(s), fetched %s", icons));
+  }
+
+  @Test
+  void shouldGetIconsWithoutPager() throws IOException {
+
+    String fileResourceId1 = createFileResource();
+    createCustomIcon(fileResourceId1, keyword, iconKey1);
+
+    String fileResourceId2 = createFileResource();
+    createCustomIcon(fileResourceId2, keyword, iconKey2);
+
+    String fileResourceId3 = createFileResource();
+    createCustomIcon(fileResourceId3, keyword, iconKey3);
+
+    JsonObject iconResponse = GET("/icons?type=CUSTOM&paging=false").content(HttpStatus.OK);
+
+    JsonPage pager = iconResponse.get("pager", JsonPage.class);
+
+    JsonList<JsonIcon> icons = iconResponse.getList("icons", JsonIcon.class);
+
+    assertHasNoMember(iconResponse, "pager");
+    assertHasNoMember(pager, "total");
```

Comment:
If it has no pager then it has not total as the property is `pager.total`. Please remove this one.